inc/azure_c_util/thandle.h

```diff
@@ -411,6 +416,34 @@ void THANDLE_MOVE(T)(THANDLE(T) * t1, THANDLE(T) * t2 )
 
+/*given a previous type T, this introduces THANDLE_INITIALIZE_MOVE_MACRO macro to move a handle (*t1=t2, *t2=NULL)*/
+#define THANDLE_INITIALIZE_MOVE_MACRO(T)                                                                                                                            \
+void THANDLE_INITIALIZE_MOVE(T)(THANDLE(T) * t1, THANDLE(T) * t2 )                                                                                                  \
+{                                                                                                                                                                   \
+    if(                                                                                                                                                             \
+        /*Codes_SRS_THANDLE_01_001: [ If t1 is NULL then THANDLE_INITIALIZE_MOVE shall return. ]*/                                                                  \
+        (t1 == NULL) ||                                                                                                                                             \
+        /*Codes_SRS_THANDLE_01_002: [ If t2 is NULL then THANDLE_INITIALIZE_MOVE shall return. ]*/                                                                  \
+        (t2 == NULL)                                                                                                                                                \
+    )                                                                                                                                                               \
+    {                                                                                                                                                               \
+        LogError("invalid argument THANDLE(" MU_TOSTRING(T) ") * t1=%p, THANDLE(" MU_TOSTRING(T) ") t2=%p", t1, t2 );                                               \
+    }                                                                                                                                                               \
+    else                                                                                                                                                            \
+    {                                                                                                                                                               \
+        if (*t2 == NULL)                                                                                                                                            \
+        {                                                                                                                                                           \
+            /*Codes_SRS_THANDLE_01_003: [ If *t2 is NULL then THANDLE_INITIALIZE_MOVE shall THANDLE_DEC_REF *t1, set *t1 to NULL and return. ]*/                    \
```

Comment:
where's teh DEC_REF?

Reply:
Spec fixed, we should not dec ref anything.

---
In reply to: [470884242](https://github.com/Azure/azure-c-util/pull/60#discussion_r470884242) [](ancestors = 470884242)